ORStools/gui/directions_gui.py

```diff
@@ -31,11 +31,44 @@
 
 from PyQt5.QtWidgets import QCheckBox
 
-from ORStools.utils import transform
+from ORStools.utils import transform, exceptions
+
+
+def _get_avoid_polygons(layer):
+    """
+    Extract polygon geometries from the selected polygon layer.
+
+    :param layer: The polygon layer
+    :type layer: QgsMapLayer
+    :returns: GeoJSON object
+    :rtype: dict
+    """
+    polygons = None
+    transformer = transform.transformToWGS(layer.sourceCrs())
+    features = layer.getFeatures()
+
+    # iterate over all other features
+    for feature in features:
+        if feature.hasGeometry():
+            geom = feature.geometry()
+            geom.transform(transformer)
+            if polygons is None:
+                polygons = geom
+            else:
+                polygons = polygons.combine(geom)
+
+    if not polygons:
+        raise exceptions.GenericServerError(
```

Comment:
I don't think the code should raise an exception here.
Either we want this to fail w/ an API Error, or we want to warn our users in the same way that we do when points or the api-key is missing.

Reply:
Changed to info popup and request abortion strategy